Submodels/MTFLO_call.py

```
@@ -1,20 +1,183 @@
+        self.process.stdin.flush()
+
+         # Check that MTFLO has closed successfully 
+        if self.process.poll() is None:
+            try:
+                self.process.wait(timeout=1)
+            
+            except subprocess.TimeoutExpired:
+                self.process.kill()
+                raise OSError("Something went wrong in the MTFLO call. \
+                            MTFLO was not closed following end of file generation. \
+                            Run terminated.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)